authorization.md

```diff
@@ -0,0 +1,99 @@
+# Authorization
+
+KubeVirt authorization is performed using Kubernete's Resource Based
+Authorization Control system (RBAC). RBAC allows cluster admins to grant
+access to cluster resources by binding RBAC roles to users.
+
+For example, an admin creates an RBAC role that represents the permissions
+required to create a VirtualMachine. The admin can then bind that role to users
+in order to grant them the permissions required to launch a VirtualMachine.
+
+With RBAC roles, admins can grant users targeted access to various KubeVirt
+features.
+
+## KubeVirt Default RBAC ClusterRoles
+
+KubeVirt comes with a set of predefined RBAC ClusterRoles that can be used to
+grant users permissions to access KubeVirt Resources.
+
+### Default View Role
+
+The **kubevirt.io:view** ClusterRole gives users permissions to view all
+KubeVirt resources in the cluster. The permissions to create, delete, modify
+or access any KubeVirt resources beyond viewing the resource's spec are not
+included in this role. This means a user with this role could see that a 
+VirtualMachine is running, but neither shutdown nor gain access to that
+VirtualMachine via console/VNC.
+
+### Default Edit Role
+
+The **kubevirt.io:edit** ClusterRole gives users permissions to modify all
+KubeVirt resources in the cluster. For example, a user with this role can
+create new VirtualMachines, delete VirtualMachines, and gain access to both
+console and VNC.
+
+### Default Admin Role
+
+The **kubevirt.io:admin** ClusterRole grants users full permissions to all
+KubeVirt resources, including the ability to delete collections of resources.
+
+## Default Config Role
+
+The **kubevirt.io:config** Role grants users access to view and modify the
```

Comment:
I really wonder if we are sure about this. A real cluster-admin has access to it anyway.

Reply:
My thought is that someone who doesn't have permission to install KubeVirt might need permissions to modify the runtime config to enable things like debugging.

Reply:
Is this really a common enough situation? This user, who is not an admin, would change the behaviour for all namespaces when changin the map. That sounds like you really need to know what you are doing.  An admin could still always create the access permission by hand if needed, i have the feeling that this is not a comnon enough scenario to add an extra role for it.

Reply:
@rmohr how about I role the config map permission into the admin role?

If a ClusterRoleBinding is used with the admin ClusterRole, then the user will get the ability to access the kubevirt-config no matter what namespace it is in.  If a RoleBinding is used, the admin will only get access to kubevirt-config if the config exists in the target namespace.

That way we don't have conifg specific ClusterRole.